lib/response.ts

```diff
@@ -0,0 +1,51 @@
+import { hasOwnProperty } from './utils/fn';
+
+export class Response {
+  body: any;
+  headers: any;
+  status: number;
+  constructor(options: any = {}) {
+    this.body = options.body || {};
+    this.headers = {};
+    this.status = 200;
+
+    // Store the headers in lower case.
+    for (const field of Object.keys(options.headers || {})) {
+      if (hasOwnProperty(options.headers, field)) {
+        this.headers[field.toLowerCase()] = options.headers[field];
+      }
+    }
+
+    // Store additional properties of the response object passed in
```

Comment:
```suggestion
    // Store additional properties of the response object passed in.
```

Reply:
(although it should be obvious from the code what this is doing).

Reply:
Yes the code is obvious, but it is actually important to comment this, because we could also filter all additional properties out of the response object